tests/unit/s3/test_transfer.py

```diff
@@ -148,6 +148,17 @@ def test_upload_file(self):
             'smallfile', 'bucket', 'key', extra_args, None
         )
 
+    def test_upload_file_via_path(self):
+        from pathlib import Path
+
+        extra_args = {'ACL': 'public-read'}
+        self.transfer.upload_file(
+            Path('smallfile'), 'bucket', 'key', extra_args=extra_args
```

Comment:
I simplify the basis by returning the `fspath` of the Path-like object which is a string. If there is an issue with the string, it would occur with or without this feature given that the original functionality only accepts a string. If the user messes up their Path-like object's path somehow, I think that's up to them.

Reply:
I understand that with your current code there is no difference between integration-testing with a path-like object and integration testing with a string. However, one purpose of tests is to have them fail when a future code contributor changes the implementation in a way that you did not expect. That would be the purpose of having one integration test that uses a path-like object.

Reply:
I can add those additional unit tests for path-like objects, but I will also add them as regular strings as well.